sycl/source/detail/config.cpp

```diff
@@ -62,36 +63,70 @@ void readConfig() {
   }
 
   if (File.is_open()) {
-    // TODO: Use max size from macro instead of 256
-    char Key[MAX_CONFIG_NAME] = {0}, Value[256] = {0};
+    char Key[MAX_CONFIG_NAME] = {0}, Value[MAX_CONFIG_VALUE] = {0};
+    std::string BufString;
+    std::size_t Position = std::string::npos;
     while (!File.eof()) {
-      // Expected fromat:
+      // Expected format:
       // ConfigName=Value\r
       // ConfigName=Value
       // TODO: Skip spaces before and after '='
-      File.getline(Key, sizeof(Key), '=');
-      if (File.fail()) {
-        // Fail to process the line. Skip it completely and try next one.
-        // Do we want to restore here? Or just throw an exception?
+      std::getline(File, BufString);
+      if (File.fail() && !File.eof()) {
+        // Fail to process the line.
         File.clear(File.rdstate() & ~std::ios_base::failbit);
         File.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+        throw sycl::exception(
+            make_error_code(errc::runtime),
+            "An error occurred while attempting to read a line");
+      }
+      // Skip lines with a length = 0 | which have a comment | don't have "="
```

Comment:
will this break the following case?

```VARIABLE=Values # some comment for the value?```
It looks like valid case.

Reply:
This line will be skipped because it is an unexpected format.

Reply:
But my everyday experience tells me that it should not. Can you fix that?

Reply:
Fixed.